llvm/lib/Target/SBF/SBFISelLowering.cpp

```diff
@@ -63,7 +63,8 @@ SBFTargetLowering::SBFTargetLowering(const TargetMachine &TM,
   // Compute derived properties from the register classes
   computeRegisterProperties(STI.getRegisterInfo());
 
-  setStackPointerRegisterToSaveRestore(SBF::R11);
+  if (Subtarget->getHasDynamicFrames())
+    setStackPointerRegisterToSaveRestore(SBF::R10);
 
```

Comment:
are you sure this will work?.. you set the sp register to save restore only if subtarget has dynamic frames.  If the assumption here is that dynamic frames are enabled by default, then why do you need this "if"?..

Reply:
If we don't have dynamic frames, R10 is not a stack pointer, but instead a frame pointer.

Reply:
yes, but you also seem to remove saving the r11 sp for the case when the target doesn't have dynamic frames.  So before your change r11 is sp and it is set to save/restore. With your change r10 is sp when the subtarget has dynamic frames and it is set to save/restore, but if subtarget doesn't have dynamic frames no sp is set to save/restore?